trace/kafka/kafka.go

```diff
@@ -31,6 +36,15 @@ func NewJSONMessage(t string, d interface{}) (*Message, error) {
 	return &Message{topic: t, body: b}, nil
 }
 
+// NewJSONMessageWithKey creates a new message with a JSON encoded body and a message key
+func NewJSONMessageWithKey(t string, d interface{}, k *string) (*Message, error) {
```

Comment:
Since you want a message with key, key cannot be nil. We should then check if `key == ""` and return an error.
```suggestion
func NewJSONMessageWithKey(t string, d interface{}, k string) (*Message, error) {
```

Reply:
Sorry for not seeing this earlier...